lazy-pages/src/lib.rs

```diff
@@ -94,18 +168,17 @@ pub unsafe fn init() -> bool {
         return true;
     }
 
-    if !LAZY_PAGES_INFO.with(|x| x.borrow().is_empty()) {
-        log::error!("Lazy pages info must be empty before initialization");
-        return false;
-    }
-
-    if !WASM_MEM_BEGIN.with(|x| *x.borrow() == 0) {
-        log::error!("Wasm mem begin must be 0 before initialization");
-        return false;
-    }
+    let ctx_is_default = LAZY_PAGES_CONTEXT.with(|ctx| {
+        let ctx = ctx.borrow();
+        ctx.lazy_pages.is_empty()
```

Comment:
```suggestion
        ctx == LazyPagesExecutionContext::default()
```

Reply:
This will create new object of lazy pages context

Reply:
It will but only once during the first initialization

Reply:
> One more solution - wrap your stuff into some method and code become more clear

But you still need to not forget compare new fields

Reply:
Ok done as more safe